### PR TITLE
Refactor CSS to use SCSS, with autoprefixer and minification

### PR DESCRIPTION
**Pull request by juliaevk**
Per issue #1978

Could probably make better use of mixins and variables, but this PR should take care of the heavy lifting for a better front end build process!

### CODE REVIEW

File: web/src/main.js [hunk 1]
```diff
@@ -1,4 +1,7 @@
-'use strict';
+console.log('test');
```
Comment (tamarau) [inline web/src/main.js]: rogue console.log :)

File: web/src/main.js [hunk 2]
```diff
@@ -13,6 +16,8 @@ import FAQ from './components/faq';
 import TimeSlider from './components/timeslider';
 import LanguageSelect from './components/languageselect';
 
+'use strict';
```
Comment (tamarau) [inline web/src/main.js]: By convention 'use strict' should be added at the beginning of a script/function

**Comment by tamarau** [comment]
I'm not going to officially review it (as it's not my project) but I left two comments, but the rest looks fine to me :)

**Comment by juliaevk** [comment]
Thanks @tamarau! Fixed.

**Comment by ovbm** [comment]
Wow, that's amazing, will review ASAP.

File: web/src/scss/_base.scss [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+
+.large-screen-hidden {
+    @include respond-to('medium-up') {
+        display: none;
+    }
+}
+
+.small-screen-hidden {
+    @include respond-to('small') {
+        display: none;
```
Comment (ovbm) [inline web/src/scss/_base.scss]: !important flag is needed, else the left-panel contents won't properly switch from country list to detail.
```suggestion
        display: none !important;
```

File: web/src/scss/_base.scss [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+    height: 99.9%; /* Due to a bug (box sizing?), we can't set 100% (the map gets undraggable on Safari iPhone 5S) */
+}
+
+.all-screens-hidden {
+    display: none !important;
+}
+
+.large-screen-hidden {
+    @include respond-to('medium-up') {
+        display: none;
```
Comment (ovbm) [inline web/src/scss/_base.scss]: !important flag is needed, else the left-panel contents won't properly switch from country list to detail.
```suggestion
        display: none !important;
```

File: web/src/scss/styles.scss [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+@import "content/map/map";
+@import "content/map/controls";
+@import "content/map/legend";
+@import "content/panel/country-panel";
+@import "content/panel/time-slider";
+@import "content/panel/zone-list";
+
+@import "components/faqs";
+@import "components/messaging";
+@import "components/modal";
```
Comment (ovbm) [inline web/src/scss/styles.scss]:
```suggestion
@import "components/modal";
@import "components/meta";
```

File: web/src/scss/layout/_header.scss [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+    & > div,
+    & > span {
+        vertical-align: middle;
+    }
+
+    a {
+        color: $white;
+        transition: color 0.4s;
+
+        &::hover {
```
Comment (ovbm) [inline web/src/scss/layout/_header.scss]:
```suggestion
        &:hover {
```

File: web/src/scss/layout/_header.scss [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+            color: #4178AC;
```
Comment (ovbm) [inline web/src/scss/layout/_header.scss]:
```suggestion
            color: #4178AC !important;
```

**Comment by ovbm** [comment]
Great work! I found a couple of small issues. We'll have to have someone else look over it as well probably, just to be sure we catch all bugs before merging (it's quite a lot :P). Could you make sure that the style changes of the checkbox that were just merged, are also included: https://github.com/tmrowco/electricitymap-contrib/pulls?utf8=%E2%9C%93&q=is%3Apr+is%3Aclosed

**Comment by juliaevk** [comment]
@ovbm For sure - will do ASAP!

**Comment by corradio** [comment]
@juliaevk friendly ping ;-)

**Comment by juliaevk** [comment]
Hi Olivier, apologies for the delay - updates in progress!

**Comment by corradio** [comment]
@juliaevk sorry to ping again, but I believe we should resolve conflicts and merge quite quickly to avoid any troubles.
I'm afraid styles.css already has started diverging.

**Comment by ovbm** [comment]
If you are ok with it, I can take over- in case you are time constrained.

**Comment by ovbm** [comment]
I'm going to close this as it's been merged via #2115 
Thanks a lot for the initiative!

**Pull request closed without merging**